columnphysics/icepack_snow.F90

```diff
@@ -51,33 +60,96 @@ subroutine icepack_init_snow
 
       ! local variables
 
+      integer (kind=int_kind) :: n
+
       character (len=*),parameter :: subname='(icepack_init_snow)'
 
       !-----------------------------------------------------------------
       ! Snow metamorphism lookup table
       !-----------------------------------------------------------------
 
+      ! if snw_aging_table = 'snicar'
+      ! best-fit parameters are read from a table (netcdf)
+      ! snowage_tau, snowage_kappa, snowage_drdt0
+      !  11 temperatures from 223.15 to 273.15 K by 5.0
+      !  31 temperature gradients from 0 to 300 K/m by 10
+      !   8 snow densities from 50 to 400 kg/m3 by 50
+
+      ! if snw_aging_table = 'test'
+      ! for testing Icepack without netcdf,
+      ! use a subsampled, hard-coded table
+      !   5 temperatures from 243.15 by 5.0 K
+      !   5 temperature gradients from 0 to 40 K/m by 10
+      !   1 snow density at 300 kg/m3
+
+      ! if snw_aging_table = 'file'
+      ! all data has to be passed into icepack_parameters
+
+      ! tcraig, should be 223.15 and 243.15 below
```

Comment:
Ah, this is what's different with when reading the file.  I think these should be fixed here, now.
If we do that, then is there any reason to have the 'snicar' option?  Speed?

Reply:
Actually, the documentation isn't quite correct.  The 11 temperatures goe from 223 to 273 by 5 while on the actual snicar file they go from 223.15 to 273.15 by 5.  That is the "bug" in the hardwired "snicar" values.  

The only reason to keep snicar is if there are table input file that do not have the 1D data on them.  Or if we want to preserve the old 1D values which may not be correct.

Reply:
@njeffery
@apcraig turned up an inconsistency between the snow aging file that I gave him (which was the most recent one on LANL IC machines, snicar_drdt_bst_fit_60_c04262019.nc), where the code uses 223.0 as the starting T index but the file has 223.15.  My inclination is to fix this in the code, but I want to make sure that the file I'm using is correct.

Alternatively, we can dispense with the index calculation in the code completely, and just read it from the file.  That's much better from a software perspective.  Thoughts?

Reply:
I double checked the snicar file and it's correct. Also, the ELM code uses an integer 223 in this expression, not 223.15.